SkEditor/Utilities/Syntax/SyntaxLoader.cs

```diff
@@ -10,129 +10,209 @@
+        var directories = Directory.GetDirectories(SyntaxFolder).ToList();
+        foreach (var directory in directories)
         {
-            Syntaxes.Add(Path.GetFileName(file));
-        });
+            try
+            {
+                FileSyntax syntax = await FileSyntax.LoadSyntax(directory);
+                if (syntax.Config.Extensions.Length == 0) 
+                    return;
+                
+                RegisterSyntax(syntax);
+            }
+            catch (Exception e)
+            {
+                await ApiVault.Get().ShowMessageWithIcon("Error", $"Failed to load syntax {directory}\n\n{e.Message}\n{e.StackTrace}", new SymbolIconSource() { Symbol = Symbol.ImportantFilled },
+                    primaryButton: false);
+            }
+        }
+    }
 
-        Directory.GetFiles(Path.Combine(SyntaxFolder, "Other Languages")).Where(file => Path.GetExtension(file).Equals(".xshd")).ToList().ForEach(file =>
+    private static void RegisterSyntax(FileSyntax syntax)
+    {
+        FileSyntaxes.Add(syntax);
+        foreach (string extension in syntax.Config.Extensions)
         {
-            string[] extensions = Path.GetFileNameWithoutExtension(file).Split('-');
-            OtherLanguageSyntaxes.Add(Path.GetFileName(file), extensions.ToList());
-        });
+            if (SortedFileSyntaxes.ContainsKey(extension))
+                SortedFileSyntaxes[extension].Add(syntax);
+            else SortedFileSyntaxes.Add(extension, [syntax]);
+        }
+    }
 
-        string currentSyntax = ApiVault.Get().GetAppConfig().CurrentSyntax;
+    public static void CheckConfiguredFileSyntaxes()
+    {
+        foreach (var fileSyntax in FileSyntaxes)
+        { 
+            if (!ApiVault.Get().GetAppConfig().FileSyntaxes.ContainsKey(fileSyntax.Config.LanguageName))
+            {
+                ApiVault.Get().GetAppConfig().FileSyntaxes.Add(fileSyntax.Config.LanguageName, fileSyntax.Config.FullIdName);
+            }
+        }
+    }
 
-        if (!Syntaxes.Contains(currentSyntax))
+    public static async Task<FileSyntax> LoadSyntax(string folder)
+    {
+        var fileSyntax = await FileSyntax.LoadSyntax(folder);
+        RegisterSyntax(fileSyntax);
+        return fileSyntax;
+    }
+    
+    public static async Task<bool> UnloadSyntax(string folder)
+    {
+        var fileSyntax = await FileSyntax.LoadSyntax(folder);
+        FileSyntaxes.RemoveAll(x => x.Config.FullIdName.Equals(fileSyntax.Config.FullIdName));
+        foreach (string extension in fileSyntax.Config.Extensions)
         {
-            SetDefaultSyntax();
+            if (SortedFileSyntaxes.ContainsKey(extension)) SortedFileSyntaxes[extension].Remove(fileSyntax);
+
+            if (ApiVault.Get().GetAppConfig().FileSyntaxes.ContainsKey(fileSyntax.Config.LanguageName))
+            {
+                ApiVault.Get().GetAppConfig().FileSyntaxes.Remove(fileSyntax.Config.LanguageName);
```

Comment:
ContainsKey isn't needed, if it isn't in the dictionary the Remove method will just return false

Reply:
Oh gotcha, didn't knew that